app/src/main/java/com/tvtracker/service/MediaService.kt

```diff
@@ -16,17 +17,23 @@ class MediaService : IMediaService {
         return withContext(Dispatchers.IO) {
             val service = ImdbRetrofitClientInstance.retrofitInstance?.create(IMediaItemDAO::class.java)
             val mediaItems = async {service?.searchImdb(text, type, page)}
-            val result = mediaItems.await()?.awaitResponse()?.body()
-            return@withContext result
+            return@withContext mediaItems.await()?.awaitResponse()?.body()
         }
     }
 
     override suspend fun searchByImdbId(imdbId: String): MediaItem? {
         return withContext(Dispatchers.IO) {
             val service = ImdbRetrofitClientInstance.retrofitInstance?.create(IMediaItemDAO::class.java)
             val mediaItems = async {service?.searchByImdbId(imdbId)}
-            val result = mediaItems.await()?.awaitResponse()?.body()
-            return@withContext result
+            return@withContext mediaItems.await()?.awaitResponse()?.body()
+        }
+    }
+
+    override suspend fun addWatchedMovie(userId: Int, movieId: Int): MoviesWatched {
+        return withContext(Dispatchers.IO) {
+            val service = ImdbRetrofitClientInstance.retrofitInstance?.create(IMediaItemDAO::class.java)
+            val mediaItems = async {service?.addWatchedMovie(movieId)}
+            return@withContext mediaItems.await()?.awaitResponse()?.body()
```

Comment:
This looks like an incomplete feature.
Are you using these anywhere?  I don't see them used.
If not in use, remove them.  Unused code is clutter, and bad documentation, which increases technical debt.
Only complete features, which meet the team's Definition of Done, should be merged to Main/Master.